components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/service/notification/NotificationTemplateManager.java

```diff
@@ -42,6 +42,24 @@ default NotificationTemplate getNotificationTemplate(String notificationChannel,
         return null;
     }
 
+    /**
+     * Return the notification template from the tenant registry which matches the given channel and template name.
+     *
+     * @param notificationChannel Notification Channel Name (Eg: SMS or EMAIL)
+     * @param templateType        Display name of the template
+     * @param locale              Locale
+     * @param tenantDomain        Tenant Domain
+     * @param applicationUuid     Application UUID
+     * @return Return {@link org.wso2.carbon.identity.governance.model.NotificationTemplate} object
+     * @throws NotificationTemplateManagerException If an error occurred while getting the notification template
+     */
+    default NotificationTemplate getNotificationTemplate(String notificationChannel, String templateType, String locale,
```

Comment:
Check if we can go with enum for notification channel. And add example value for the comment as well.

Reply:
Added the example values with https://github.com/wso2-extensions/identity-governance/pull/811/commits/58587abf4c12a0b457e48a41093beb5709daa915. Will track the enum improvement for later.